examples/lottery/src/lib.rs

```diff
@@ -0,0 +1,346 @@
+                                prev_winning_tickets.push(winning_ticket_id);
+
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                self.clear_storage(&lottery_name);
+
+                Ok(())
+            }
+        }
+    }
+
+    #[view]
+    fn status(&self, lottery_name: Vec<u8>) -> Status {
+        // Ticket_price 0 is invalid. Using the fact that memory is initialized to 0 by default.
+        if self.get_ticket_price(&lottery_name) == 0 {
+            return Status::Inactive;
+        }
+        if self.get_block_timestamp() > self.get_deadline(&lottery_name) || 
+            *self.get_mut_tickets_left(&lottery_name) == 0 {
+            return Status::Ended;
+        }
+
+        return Status::Running;
+    }
+
+    fn random(&self, seed: u64) -> u32 {
+        let hash_array = self.sha256(&seed.to_be_bytes());
+        let first_byte = (hash_array[28] as u32) << 24;
+        let second_byte = (hash_array[29] as u32) << 16;
+        let third_byte = (hash_array[30] as u32) << 8;
+        let fourth_byte = hash_array[31] as u32;
+
+        return first_byte | second_byte | third_byte | fourth_byte;
+    }
+
+    fn clear_storage(&self, lottery_name: &Vec<u8>) {
+        let name_len_vec = &(lottery_name.len() as u32).to_be_bytes().to_vec();
+        let temp = [&name_len_vec[..], &lottery_name[..]].concat(); // "temporary value dropped" otherwise
+        let appended_name_in_key = temp.as_slice();
+
+        self.storage_store(&["deadline".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+        self.storage_store(&["ticketPrice".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+        self.storage_store(&["ticketsLeft".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+        self.storage_store(&["prizePool".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+        self.storage_store(&["maxEntriesPerUser".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+        self.storage_store(&["prizeDistribution".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+        self.storage_store(&["whitelist".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+
+        let last_ticket = self.get_mut_current_ticket_number(lottery_name);
+
+        for i in 0..*last_ticket {
+            let addr = self.get_ticket_holder(lottery_name, i);
+            let key_ticket_holder = ["ticketHolder".as_bytes(),
+                appended_name_in_key, &i.to_be_bytes()].concat();
+            let key_number_of_entries = ["numberOfEntriesForUser".as_bytes(),
+                appended_name_in_key, addr.as_bytes()].concat();
+
+            self.storage_store(&key_ticket_holder, &[0u8; 0]);
+            self.storage_store(&key_number_of_entries, &[0u8; 0]);
+        }
+
+        self.storage_store(&["currentTicketNumber".as_bytes(), appended_name_in_key].concat(), &[0u8; 0]);
+    }
+
+    #[storage_set("deadline")]
```

Comment:
all set and gets can be moved to another file

Reply:
I think they can stay here, especially if some of them get coalesced into a single state

Reply:
but the serialization/deserialization and the state structure would be nice to have in another file. The only reason I put it in the same file in the tutorial was so people can easily copy-paste the code at the end of the tutorial